tests/cbmc/proofs/s2n_mem_cleanup/s2n_mem_cleanup_harness.c

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "api/s2n.h"
+#include "utils/s2n_blob.h"
+#include "utils/s2n_mem.h"
+
+#include <assert.h>
+#include <cbmc_proof/cbmc_utils.h>
+#include <cbmc_proof/make_common_datastructures.h>
+#include <cbmc_proof/proof_allocators.h>
+
+void s2n_mem_cleanup_harness() {
+
+    /* Non-deterministically set initialized (in s2n_mem) to true. */
+    if(nondet_bool()) {
+        s2n_mem_init();
+    }
+
+    bool old_init = s2n_mem_is_init();
+
+    if (s2n_mem_cleanup() == S2N_SUCCESS) {
+        assert(old_init);
+        assert(!s2n_mem_is_init());
+    }
```

Comment:
We could assert that
```c
assert(!old_init);
```
right? What would the case be in failure? Or that is not an option?

Reply:
If old_init is false to begin with, then s2n_mem_cleanup will fail. However, there are other ways that s2n_mem_cleanup can fail when old_init is true, so there's no specific failure case for it.